# Code

Packages Needed:

In [1]:
import pandas as pd
import requests
import csv
from collections import Counter

### 1) Import the survey.

In [2]:
csv_url = 'https://gist.githubusercontent.com/clbarnes/3d4cef38806672535a37a7db3c06c0ad/raw/fc69fe3c7b198eb062246027e7c1547c804b5b4e/HCMST_ver_3.04.csv'
req = requests.get(csv_url)
url_content = req.content
couplesData = open('couplesData.csv', 'wb')

couplesData.write(url_content)
couplesData.close()

The next chunk of code describes the shape and reveals the first five lines of data.

In [3]:
couples = pd.read_csv('couplesData.csv', sep =',', header = 0, parse_dates = [0], low_memory=False)
print(couples.shape)
couples.head()

(4002, 388)


Unnamed: 0  caseid_new  weight1  weight2  ppage ppagecat ppagect4  \
0          0       22526     4265   4265.0     52    45-54    45-59   
1          1       23286    16485  16485.0     28    25-34    18-29   
2          2       25495    52464      NaN     49    45-54    45-59   
3          3       26315     4575   4575.0     31    25-34    30-44   
4          4       27355    12147      NaN     35    35-44    30-44   

                                              ppeduc  \
0                                   bachelors degree   
1                                     masters degree   
2  high school graduate - high school diploma or ...   
3                                   associate degree   
4  high school graduate - high school diploma or ...   

                      ppeducat               ppethm  ... w3_mbtiming_year  \
0  bachelor's degree or higher             hispanic  ...              NaN   
1  bachelor's degree or higher  white, non-hispanic  ...              NaN   
2                  high school  black, non-hispanic  ...              NaN   
3                 some college  white, non-hispanic  ...              NaN   
4                  high school  white, non-hispanic  ...              NaN   

  w3_mbtiming_month  w3_q5 w3_q6                       w3_q7  \
0               NaN    yes   yes  no, did not marry [xNameP]   
1               NaN    NaN   NaN                         NaN   
2               NaN    NaN   NaN                         NaN   
3               NaN    NaN   NaN                         NaN   
4               NaN    NaN   NaN                         NaN   

                                               w3_q8 w3_q9 w3_q10  \
0  No, we have not gotten a domestic partnership ...   NaN    NaN   
1                                                NaN   NaN    NaN   
2                                                NaN   NaN    NaN   
3                                                NaN   NaN    NaN   
4                                                NaN   NaN    NaN   

  w3_nonmbtiming_year w3_nonmbtiming_month  
0                 NaN                  NaN  
1                 NaN                  NaN  
2                 NaN                  NaN  
3                 NaN                  NaN  
4                 NaN                  NaN  

[5 rows x 388 columns]

### 2) Clean up our data.

The purpose of the next chunk of code is to delete any columns that are not useful.  For instance, we deleted "weight1" and "weight2" because those values are only useful for the people who conducted the survey.

In [4]:
clean_couples = couples.drop(['Unnamed: 0', 'caseid_new', 'weight1', 'weight2','pphhhead', 'hhinc', 'ppreg9', 'pprent', 
                              'ppt01', 'ppt1317', 'ppt18ov', 'ppt25', 'ppt612', 'ppnet', 'papglb_friend', 'papevangelical', 
                              'ppppcmdate_yrmo', 'pppadate_yrmo', 'pphhcomp11_member2_age', 'pphhcomp11_member3_age', 'pphhcomp11_member4_age', 
                              'pphhcomp11_member5_age', 'pphhcomp11_member6_age', 'pphhcomp11_member7_age', 'pphhcomp11_member8_age', 
                              'pphhcomp11_member9_age', 'pphhcomp11_member10_age', 'pphhcomp11_member11_age', 
                              'pphhcomp11_member12_age','pphhcomp11_member13_age', 'pphhcomp11_member14_age', 
                              'pphhcomp11_member15_age', 'pphhcomp11_member2_gender', 'pphhcomp11_member3_gender', 'pphhcomp11_member4_gender', 
                              'pphhcomp11_member5_gender', 'pphhcomp11_member6_gender','pphhcomp11_member7_gender', 
                              'pphhcomp11_member8_gender', 'pphhcomp11_member9_gender', 'pphhcomp11_member10_gender', 
                              'pphhcomp11_member11_gender', 'pphhcomp11_member12_gender', 'pphhcomp11_member13_gender', 
                              'pphhcomp11_member14_gender', 'pphhcomp11_member15_gender', 'pphhcomp11_member2_relationship', 
                              'pphhcomp11_member3_relationship', 'pphhcomp11_member4_relationship', 'pphhcomp11_member5_relationship', 
                              'pphhcomp11_member6_relationship', 'pphhcomp11_member7_relationship', 'pphhcomp11_member8_relationship', 
                              'pphhcomp11_member9_relationship', 'pphhcomp11_member10_relationship', 
                              'pphhcomp11_member11_relationship', 'pphhcomp11_member12_relationship', 
                              'pphhcomp11_member13_relationship', 'pphhcomp11_member14_relationship', 
                              'pphhcomp11_member15_relationship', 'irb_consent', 'weight3', 'weight4', 'weight5', 'weight6', 
                              'weight7', 'weight_couples_coresident', 'HCMST_main_interview_yrmo', 'duration', 'qflag', 'glbstatus', 
                              'papglb_status', 'recsource', 'q3_codes', 'q8a', 'q8b', 'q11', 'q13a', 'q13b', 'q14', 
                              'q15a1_compressed', 'q16', 'q17a', 'q17b', 'q17c', 'q17d', 'gender_attraction', 'q18b_codes', 
                              'q18a_1', 'q18a_2', 'q18a_3', 'q18a_refused', 'q18b_codes', 'q18c_codes', 'q20', 'q21a_refusal', 
                              'q21b_refusal', 'q21c_refusal', 'q21d_refusal', 'q21e', 'q21e_refusal', 'q24_codes', 'q28', 'q29', 
                              'q31_other_text_entered', 'q33_other_text_entered', 'q35_codes', 'q35_text_entered', 
                              'summary_q24_total', 'q24_P_cowork', 'q24_P_friend', 'q24_P_family', 'q24_P_sig_other', 
                              'q24_P_neighbor', 'q24_btwn_I_cowork', 'q24_btwn_I_friend', 'q24_btwn_I_family', 'q24_btwn_I_sig_other', 
                              'q24_btwn_I_neighbor', 'q24_fam_sister_active', 'q24_fam_brother_active', 'q24_fam_mother_active', 
                              'q24_fam_father_active', 'q24_fam_other_active', 'q24_fam_cousins_active', 'q24_fam_aunt_niece_active', 
                              'q24_fam_uncle_nephew_active', 'q24_fam_grandmother_active', 'q24_fam_grandfather_active', 
                              'q24_fam_sister_passive', 'q24_fam_brother_passive', 'q24_fam_mother_passive', 'q24_fam_father_passive', 
                              'q24_fam_other_passive', 'q24_fam_cousins_passive', 'q24_fam_aunt_niece_passive', 
                              'q24_fam_uncle_nephew_passive', 'q24_fam_grandmother_passive', 'q24_fam_grandfather_passive', 
                              'q24_fam_female', 'q24_fam_male', 'distancemoved_10mi', 'marrynotreally', 'marrycountry', 
                              'civilnotreally', 'partner_deceased', 'partner_religion_reclassified', 'partner_religion_child_reclass', 
                              'own_religion_child_reclass', 'either_internet', 'either_internet_adjusted', 
                              'potential_partner_gender_recodes', 'alt_partner_gender', 'how_long_ago_first_met', 
                              'how_long_ago_first_romantic', 'how_long_ago_first_cohab', 'how_long_ago_first_met_cat', 
                              'how_long_relationship', 'respondent_religion_at_16', 'respondent_relig_16_cat', 
                              'partner_religion_at_16', 'partner_relig_16_cat', 'married', 'respondent_yrsed', 'partner_yrsed', 
                              'respondent_race', 'partner_race', 'home_country_recode', 'US_raised', 'partner_mom_yrsed', 
                              'respondent_mom_yrsed', 'relationship_quality', 'coresident', 'pp2_afterp1', 'pp2_pphhhead', 
                              'pp2_ppincimp', 'pp2_ppmsacat', 'pp2_ppeducat', 'pp2_respondent_yrsed', 'pp2_ppreg9', 'pp2_pprent', 
                              'pp2_ppt01', 'pp2_ppt1317', 'pp2_ppt18ov', 'pp2_ppt25', 'pp2_ppt612', 'pp2_ppnet', 'pp2_ppcmdate_yrmo', 
                              'pp_igdr1', 'pp_ieduc1', 'pp2_igdr2', 'pp2_ieduc2', 'w2_deceased', 'w2_multiname', 'w2_panelstat', 
                              'w2_donotcontact', 'w2_assigned', 'w2_f1complete', 'w2_HCMST_interview_fin_yrmo', 'w2_duration', 
                              'w2_xmarry', 'w2_xss', 'w2_source', 'w2_q4', 'w2_q7', 'w2_q8', 'w2_q10', 'w2_broke_up', 
                              'w2_days_elapsed', 'pp3_pphhhead', 'pp3_pphhsize', 'pp3_pphouse', 'pp3_ppincimp', 'pp3_ppmsacat', 
                              'pp3_pprent', 'pp3_ppreg4', 'pp3_ppreg9', 'interstate_mover_pp1_pp2', 'interstate_mover_pp2_pp3', 
                              'interstate_mover_pp1_pp3', 'pp3_ppt01', 'pp3_ppt1317', 'pp3_ppt18ov', 'pp3_ppt25', 'pp3_ppt612', 
                              'pp3_ppwork', 'pp3_ppnet', 'pp3_ppcmdate_yrmo', 'pp3_ppeduc', 'pp3_ppeducat', 'pp3_respondent_yrsed', 
                              'pp3_ppethm', 'pp3_newer', 'w2w3_combo_breakup', 'w3_broke_up', 'w3_xpartnered', 'w3_xdeceased', 
                              'w3_multiname', 'w3_xss', 'w3_xlast', 'w3_xyear', 'w3_xmonth', 'w3_xqualified', 'w3_status', 
                              'w3_complete', 'w3_source', 'w3_HCMST_interview_fin_yrmo', 'w3_days_elapsed', 'w3_duration', 
                              'w3_xmarry', 'w3_xtype', 'w3_q1', 'w3_q4', 'w3_mbtiming_year', 'w3_mbtiming_month', 'w3_q7', 'w3_q8', 
                              'w3_q10', 'w3_nonmbtiming_year', 'w3_nonmbtiming_month', 'ppagecat', 'ppagect4', 'ppeducat', 'pphouse', 
                              'ppmsacat', 'ppincimp', 'parental_approval', 'pp2_ppmarit', 'pp2_pphouse', 'pp2_pphouse', 'pp2_ppincimp', 
                              'pp2_ppeduc', 'pp2_pphhsize', 'pp2_ppwork', 'w3_q2', 'w3_q3', 'w3_q5', 'w3_q9', 'w3_q6', 'ppq14arace', 
                              'q32_internet', 'q24_school', 'q24_vacation', 'q24_bar_restaurant', 'q24_private_party'], 1)

The next two lines of code demonstrate our updated data. Notice that there is now only 106 columns instead of 388.

In [5]:
print(clean_couples.shape)
clean_couples.head()

(4002, 106)


ppage                                             ppeduc  \
0     52                                   bachelors degree   
1     28                                     masters degree   
2     49  high school graduate - high school diploma or ...   
3     31                                   associate degree   
4     35  high school graduate - high school diploma or ...   

                ppethm ppgender  pphouseholdsize              ppmarit  \
0             hispanic   female                2  living with partner   
1  white, non-hispanic   female                2  living with partner   
2  black, non-hispanic   female                4        never married   
3  white, non-hispanic     male                1        never married   
4  white, non-hispanic     male                2        never married   

    ppreg4  children_in_hh                        ppwork  \
0  midwest               0  working - as a paid employee   
1     west               0  working - as a paid employee   
2    south               1  working - as a paid employee   
3    south               0  working - as a paid employee   
4    south               0       working - self-employed   

                             pphispan  ... met_through_as_coworkers  \
0  yes, other spanish/hispanic/latino  ...                      0.0   
1                        no, i am not  ...                      0.0   
2                        no, i am not  ...                      NaN   
3                        no, i am not  ...                      0.0   
4                        no, i am not  ...                      NaN   

            pp2_ppethm pp2_ppreg4 w2_q1 w2_q2 w2_q3 w2_q5 w2_q6        w2_q9  \
0             hispanic    midwest   NaN   NaN   NaN   yes   yes          NaN   
1  white, non-hispanic       west   yes   yes   NaN   NaN   NaN          NaN   
2  black, non-hispanic      south   NaN   NaN   NaN   NaN   NaN          NaN   
3  white, non-hispanic      south   NaN   NaN   NaN    no   NaN  we broke up   
4  white, non-hispanic      south   NaN   NaN   NaN   NaN   NaN          NaN   

           pp3_ppmarit  
0  Living with partner  
1  Living with partner  
2        Never married  
3        Never married  
4        Never married  

[5 rows x 106 columns]

Next, we are updating all of the column names to make more sense.

In [6]:
clean_couples.rename(columns=({'ppage': 'Age', 'ppeduc': 'Education Level', 'ppethm': 'Race', 'ppgender': 'Gender', 
                               'pphouseholdsize': 'householdSize', 'ppmarit': 'maritalStatus', 'ppreg4': 'Region', 
                               'children_in_hh': 'kidsInHH', 'ppwork': 'employmentStatus', 'pphispan': 'Hispanic', 
                               'pprace_white': 'White', 'pprace_black': 'Black', 'pprace_nativeamerican': 'nativeAmerican', 
                               'pprace_asianindian': 'asainIndian', 'pprace_chinese': 'Chinese', 'pprace_filipino': 'Filipino', 
                               'pprace_japanese': 'Japanese', 'pprace_korean': 'Korean', 'pprace_vietnamese': 'Vietnamese', 
                               'pprace_otherasian': 'otherAsainEthnicity', 'pprace_hawaiian': 'Hawaiian', 
                               'pprace_guamanian': 'Guamanian', 'pprace_samoan': 'Samoan', 
                               'pprace_otherpacificislander': 'otherPacificIslander', 'pprace_someotherrace': 'someOtherRace', 
                               'pppartyid3': 'politicalParty', 'papreligion': 'Religion', 's1': 'Married', 's1a': 'LGBTQ+', 
                               's2': 'romanticPartner', 'q4': 'partnersGender', 'q5': 'sameGenderAsPartner', 
                               'q6a': 'partnerHispanic', 'q6b': 'partnersRace', 'q7a': 'partnerChristian', 
                               'q7b': 'partnersReligion', 'q9': 'partnersAge', 'q10': 'partnersEducationLevel', 
                               'q12': 'partnersPoliticalParty', 'q19': 'livingWithPartner', 'q21a': 'ageMetPartner', 
                               'q21b': 'ageRelationshipStarted', 'q21c': 'ageLivingTogether', 
                               'q21d': 'ageMarried', 'q22': 'relationshipLength', 'q23': 'whoEarnsMore', 
                               'q25': 'sameHighSchool', 'q26': 'sameCollege', 'q27': 'sameHometown', 
                               'q30': 'parentsApprove', 'q31_1': 'metAtWork', 'q31_2': 'metAtSchool', 
                               'q31_3': 'metAtChurch', 'q31_4': 'metViaInternetService', 'q31_5': 'metOnVacation', 
                               'q31_6': 'metAtBar', 'q31_7': 'metAtClubActivity', 'q31_8': 'metAtParty', 
                               'q31_9': 'metOther', 'q32': 'usedInternetSevice', 'q33_1': 'familyIntroduced', 
                               'q33_2': 'friendsIntroduced', 'q33_3': 'coworkersIntroduced', 
                               'q33_4': 'classmateIntroduced', 'q33_5': 'neighborsIntroduced', 'q33_6': 'selfIntroduced', 
                               'q33_7': 'otherIntroduced', 'q34': 'qualityOfRelationship', 'q24_met_online': 'metOnline', 
                               'q24_R_cowork': 'metAsCoworkers', 
                               'q24_R_friend': 'metThroughFriends', 'q24_R_family': 'metThroughFamily', 
                               'q24_R_sig_other': 'metThroughSigOther', 'q24_R_neighbor': 'metThroughNeighbor', 
                               'q24_college': 'metCollege', 'q24_military': 'metMilitary', 'q24_church': 'metChurch', 
                               'q24_vol_org': 'metVoluntaryOrg', 'q24_customer': 'metAsCustomer',  
                               'q24_internet_dating': 'metInternetDating', 'q24_internet_social_networking': 'metNetworking', 
                               'q24_internet_game': 'metGaming', 'q24_internet_chat': 'metInternetChat', 
                               'q24_internet_community': 'metInternetCommunity', 'q24_internet_other': 'metInternetOther', 
                               'q24_public': 'metInPublic',  
                               'q24_blind_date': 'metBlindDate',  
                               'q24_singles_service_non_internet': 'metAtSinglesServiceNonInternet', 
                               'q24_business_trip': 'metOnBusiness', 'q24_work_neighbor': 'metAsWorkNeighbors', 
                               'how_met_online': 'typeOfInternetMeeting', 'same_sex_couple': 'sameSexCouple', 
                               'age_difference': 'ageDifference', 'met_through_friends': 'metThroughFriends', 
                               'met_through_family': 'metThroughFamily', 'met_through_as_neighbors': 'metAsNeighbors', 
                               'met_through_as_coworkers': 'metThroughCoworkers', 'pp2_ppethm': 'partnersRace', 'pp2_ppreg4': 'partnersRegion', 
                               'w2_q1': 'stillMarriedToPartner', 'w2_q2': 'livingWithPartner', 'w2_q3': 'howDidMarriageEnd', 
                               'w2_q5': 'stillInRelationship', 'w2_q6': 'livingTogether', 'w2_q9': 'howDidRelationshipEnd',
                               'pp3_ppmarit' : 'relationshipStatus'}), 
    inplace=True,)
print(clean_couples.shape)

(4002, 106)


### 3) Create a new CSV file containing the cleaned up survey information.

In [7]:
clean_couples.to_csv('cleanCouplesData.csv')

### 4) Extracting data from CSV

#### How couples met vs. the result of the relationship. 

In [8]:
# Met Socially
social = []

for line in range(len(clean_couples)):

    if (clean_couples['metAtWork'][line] == 'yes' or clean_couples['metAtChurch'][line] == 'yes' or 
        clean_couples['metAtBar'][line] == 'yes' or clean_couples['metAtParty'][line] == 'yes' or
        clean_couples['selfIntroduced'][line] == 'yes' or clean_couples['metAtSchool'][line] == 'yes' or
        clean_couples['metCollege'][line] == 'yes' or clean_couples['metMilitary'][line] == 'yes' or
        clean_couples['metChurch'][line] == 'yes' or clean_couples['metVoluntaryOrg'][line] == 'yes' or
        clean_couples['metAsCustomer'][line] == 'yes' or clean_couples['metAtBar'][line] == 'yes' or
        clean_couples['metInPublic'][line] == 'yes' or clean_couples['metOnVacation'][line] == 'yes' or
        clean_couples['metOnBusiness'][line] == 'yes' or clean_couples['metAsWorkNeighbors'][line] == 'yes'):
            social.append(clean_couples['relationshipStatus'][line])

socialCount = dict(Counter(social))

In [9]:
# Met Online
online = []

for line in range(len(clean_couples)):
    
    if (clean_couples['metViaInternetService'][line] == 'yes' or clean_couples['usedInternetSevice'][line] == 'yes' or 
        clean_couples['metOnline'][line] == 'yes' or clean_couples['metInternetDating'][line] == 'yes' or
        clean_couples['metNetworking'][line] == 'yes' or clean_couples['metGaming'][line] == 'yes' or
        clean_couples['metInternetChat'][line] == 'yes' or clean_couples['metInternetCommunity'][line] == 'yes' or
        clean_couples['metInternetOther'][line] == 'yes'):
            online.append(clean_couples['relationshipStatus'][line])
            
onlineCount = dict(Counter(online))

In [10]:
# Met through mutuals
mutual = []

for line in range(len(clean_couples)):

    if (clean_couples['familyIntroduced'][line] == 'yes' or clean_couples['friendsIntroduced'][line] == 'yes' or 
        clean_couples['coworkersIntroduced'][line] == 'yes' or clean_couples['classmateIntroduced'][line] == 'yes' or
        clean_couples['neighborsIntroduced'][line] == 'yes'):
            mutual.append(clean_couples['relationshipStatus'][line])

mutualCount = dict(Counter(mutual))

In [11]:
# How couples meet and their relationship status
def howCouplesMet():
    return('Met Socially', socialCount), ('Met Through Mutuals', mutualCount), ('Met Online', onlineCount)

howCouplesMet()

(('Met Socially',
  {'Living with partner': 247,
   'Separated': 29,
   nan: 716,
   'Married': 908,
   'Never married': 200,
   'Divorced': 93,
   'Widowed': 39}),
 ('Met Through Mutuals',
  {'Living with partner': 147,
   'Never married': 118,
   'Separated': 21,
   nan: 529,
   'Married': 648,
   'Widowed': 28,
   'Divorced': 59}),
 ('Met Online',
  {'Living with partner': 26,
   nan: 38,
   'Never married': 20,
   'Divorced': 16,
   'Married': 44,
   'Widowed': 4,
   'Separated': 2}))

#### Partners ages vs. the result of the relationship.

In [12]:
# Partners ages 
age = []

for line in range(len(clean_couples)):

    if (clean_couples['Age'][line] < 18 or (18 <= clean_couples['Age'][line] <= 24) or 
         (25 <= clean_couples['Age'][line] <= 34) or  (35 <= clean_couples['Age'][line] <= 44) or 
         (45 <= clean_couples['Age'][line] <= 54) or  (55 <= clean_couples['Age'][line] <= 64) or 
         (65 <= clean_couples['Age'][line] <= 74) or  75 <= clean_couples['Age'][line]):
            age.append(clean_couples['relationshipStatus'][line])

ageCount = dict(Counter(age))

In [13]:
def couplesAge():
    return('Result of the Relationship based on Age', ageCount)

couplesAge()

('Result of the Relationship based on Age',
 {'Living with partner': 362,
  'Never married': 590,
  'Separated': 46,
  'Divorced': 297,
  nan: 1358,
  'Married': 1220,
  'Widowed': 129})

#### Partners ethnicity vs. the result of the relationship.

In [14]:
def RaceCount(race):
    wh=[]
    for line in range(len(clean_couples)):
        if(clean_couples['Race'][line] == race):
            wh.append(clean_couples['relationshipStatus'][line])
    whCount = dict(Counter(wh))
    return whCount

In [15]:
#white, non-hispanic
whCount = RaceCount('white, non-hispanic')

#black, non-hispanic
blCount = RaceCount('black, non-hispanic')

# other, non-hispanic
otCount = RaceCount('other, non-hispanic')

# hispanic
hiCount = RaceCount('hispanic')

# 2+ races, non-hispanic
raCount = RaceCount('2+ races, non-hispanic')

In [16]:
def CouplesRace():
    return('white, non-hispanic', whCount), ('black, non-hispanic', blCount), ('other, non-hispanic', otCount), ('hispanic', hiCount),('2+ races, non-hispanic', raCount)
CouplesRace()

(('white, non-hispanic',
  {'Living with partner': 272,
   'Never married': 431,
   'Separated': 20,
   'Divorced': 220,
   nan: 1024,
   'Married': 929,
   'Widowed': 111}),
 ('black, non-hispanic',
  {'Never married': 62,
   'Living with partner': 15,
   'Married': 62,
   nan: 127,
   'Widowed': 12,
   'Divorced': 28,
   'Separated': 15}),
 ('other, non-hispanic',
  {'Married': 36,
   'Never married': 27,
   nan: 36,
   'Living with partner': 15,
   'Divorced': 6,
   'Separated': 1,
   'Widowed': 1}),
 ('hispanic',
  {'Living with partner': 47,
   'Married': 152,
   'Divorced': 25,
   nan: 106,
   'Never married': 47,
   'Widowed': 4,
   'Separated': 6}),
 ('2+ races, non-hispanic',
  {'Never married': 23,
   nan: 65,
   'Married': 41,
   'Living with partner': 13,
   'Divorced': 18,
   'Separated': 4,
   'Widowed': 1}))

#### Partners education vs. the result of the relationship.

In [17]:
# Partners education 
education = []

for line in range(len(clean_couples)):

    
    if (clean_couples['Education Level'][line] == 'no formal education' or clean_couples['Education Level'][line] == '1st, 2nd, 3rd, or 4th grade' or
       clean_couples['Education Level'][line] == '5th or 6th grade' or clean_couples['Education Level'][line] == '7th or 8th grade' or 
        clean_couples['Education Level'][line] == '9th grade' or clean_couples['Education Level'][line] == '10th grade' or 
        clean_couples['Education Level'][line] == '11th grade' or clean_couples['Education Level'][line] == '12th grade no diploma' or 
        clean_couples['Education Level'][line] == 'high school graduate - high school diploma or the equivalent (ged)' or 
        clean_couples['Education Level'][line] == 'some college, no degree' or clean_couples['Education Level'][line] == 'associate degree' or 
        clean_couples['Education Level'][line] == 'bachelors degree' or clean_couples['Education Level'][line] == 'masters degree' or 
        clean_couples['Education Level'][line] == 'professional or doctorate degree' or clean_couples['Education Level'][line] == ' '):
            education.append(clean_couples['relationshipStatus'][line])
            
eduCount = dict(Counter(education))

In [18]:
# Partners education and their relationship status
def couplesEducation():
    return('Relationship Status based on Education Levels', eduCount)

couplesEducation()

('Relationship Status based on Education Levels',
 {'Living with partner': 362,
  'Never married': 590,
  'Separated': 46,
  'Divorced': 297,
  nan: 1358,
  'Married': 1220,
  'Widowed': 129})

In [24]:
correlation = clean_couples.corr()
correlation

Age  householdSize  kidsInHH  partnersAge  \
Age                     1.000000      -0.403260 -0.273624     0.896450   
householdSize          -0.403260       1.000000  0.600651    -0.322825   
kidsInHH               -0.273624       0.600651  1.000000    -0.257586   
partnersAge             0.896450      -0.322825 -0.257586     1.000000   
ageMetPartner           0.356762      -0.288367 -0.184905     0.206816   
ageRelationshipStarted  0.394389      -0.304991 -0.198683     0.246305   
ageLivingTogether       0.358793      -0.254888 -0.196916     0.212935   
ageMarried              0.259291      -0.201386 -0.154353     0.157917   
ageDifference           0.180035      -0.123306 -0.100857     0.130774   
metThroughCoworkers     0.021619      -0.015673  0.018285     0.053353   

                        ageMetPartner  ageRelationshipStarted  \
Age                          0.356762                0.394389   
householdSize               -0.288367               -0.304991   
kidsInHH                    -0.184905               -0.198683   
partnersAge                  0.206816                0.246305   
ageMetPartner                1.000000                0.931498   
ageRelationshipStarted       0.931498                1.000000   
ageLivingTogether            0.913735                0.970703   
ageMarried                   0.862087                0.921799   
ageDifference                0.360309                0.352068   
metThroughCoworkers          0.007352                0.003142   

                        ageLivingTogether  ageMarried  ageDifference  \
Age                              0.358793    0.259291       0.180035   
householdSize                   -0.254888   -0.201386      -0.123306   
kidsInHH                        -0.196916   -0.154353      -0.100857   
partnersAge                      0.212935    0.157917       0.130774   
ageMetPartner                    0.913735    0.862087       0.360309   
ageRelationshipStarted           0.970703    0.921799       0.352068   
ageLivingTogether                1.000000    0.951954       0.341286   
ageMarried                       0.951954    1.000000       0.338777   
ageDifference                    0.341286    0.338777       1.000000   
metThroughCoworkers              0.002112    0.050658       0.047750   

                        metThroughCoworkers  
Age                                0.021619  
householdSize                     -0.015673  
kidsInHH                           0.018285  
partnersAge                        0.053353  
ageMetPartner                      0.007352  
ageRelationshipStarted             0.003142  
ageLivingTogether                  0.002112  
ageMarried                         0.050658  
ageDifference                      0.047750  
metThroughCoworkers                1.000000